__tests__/integration.test.js

```diff
@@ -308,3 +307,61 @@ describe('selectUsers()', () => {
     });
   });
 });
+
+describe('selectArticles by topic query()', () => {
+  test('200:Selecting all topics with valid topic query', () => {
+    return request(app)
+      .get('/api/articles?topic=mitch')
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toHaveLength(12);
+        expect(body.articles[0]).toMatchObject({
+          article_id: expect.any(Number),
+          title: expect.any(String),
+          topic: expect.any(String),
+          body: expect.any(String),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String),
+          comment_count: expect.any(String),
+        });
+      });
+  });
+  test('200: Return [] with valid topic query but no article with this query', () => {
+    return request(app)
+      .get('/api/articles?topic=paper')
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toHaveLength(0);
+        expect(body.articles).toEqual([]);
+      });
+  });
+  test('200: Return all article with invalid topic query', () => {
+    return request(app)
+      .get('/api/articles?topic=lion')
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toHaveLength(13);
+        expect(body.articles[0]).toMatchObject({
+          article_id: expect.any(Number),
+          title: expect.any(String),
+          topic: expect.any(String),
+          body: expect.any(String),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String),
+          comment_count: expect.any(String),
+        });
+      });
+  });
+  describe('404: error handling', () => {
+    test('returns an error with invalid article endpoint', () => {
+      return request(app)
+        .get('/api/banana')
```

Comment:
this isn't specific to this endpoint so should be in it's own describe block for a generic 404/request to a path that doesn't exist in the app